### ISSUE TITLE
Better Error Response on Offsite Complete Purchase Request

### ISSUE THREAD

**Issue by pdaleramirez** [craftcms/commerce-paypal-checkout#51]
**What are you trying to do?**
When PayPal encounters an error when submitting the payment request it throws PaymentException (https://github.com/craftcms/commerce-paypal-checkout/blob/37fadcf75cd684ee64c6a49774308d6c9be1a2dd/src/gateways/Gateway.php#L401) which creates a bad user experience. 

**What's your proposed solution?**
It would be good to implement the `craft\commerce\base\RequestResponseInterface` same as how Commerce Stripe Implement returned errors

**Comment by aeu** [comment]
Hi Dale, 

Thanks for taking a look at this, but respectfully, an eCommerce site throwing a Internal Server error on a card decline is not an enhancement, it's a critical issue.  This is what our customers are seeing on card declines, which can happen for something as simple as a typo during checkout or hitting PayPal fraud protection.

Note also that when this happens the customers cart is cleared.  Also a critical issue.

![craft-exception](https://user-images.githubusercontent.com/2225377/154967061-fc6d491e-6794-4435-a76e-f5034936b9d2.png)

PayPal in particular takes a very aggressive stance on fraud protection (as they should) and card declines are not uncommon at all.  A card decline should not crash an eCommerce platform and wipe the customer cart.   

To put it another way, if I saw this during checkout, I'd take my business elsewhere.

**Comment by nfourtythree** [comment]
Hi @aeu 

Thank you for your message and extra information.

We have just pushed up an update that will fix this issue. Error messages from the API will now be returned and the user will be redirected to the payment page.

Thanks.

**Comment by aeu** [comment]
Hi @nfourtythree 

This is excellent news, thank you!

Do you have a timeline for when that release will be?

**Comment by nfourtythree** [comment]
Hi @aeu 

We have just published the `1.3.2` version of the plugin.

You will be able to update now.

Thanks!
